src/core/jwk.rs

```diff
@@ -345,34 +432,9 @@ impl CoreRsaPrivateSigningKey {
         } else if !trimmed_pem.ends_with(RSA_FOOTER) {
             return Err(format!("RSA private key must end with {}", RSA_FOOTER));
         }
-        let base64_pem = &trimmed_pem[RSA_HEADER.len()..trimmed_pem.len() - RSA_FOOTER.len()];
-        let base64_pem = Self::lax_base64_parsing(base64_pem);
-        let config = base64::STANDARD.decode_allow_trailing_bits(true);
-        let der = base64::decode_config(base64_pem, config)
-            .map_err(|_| "Failed to decode RSA private key body as base64".to_string())?;
-
-        let key_pair = ring_signature::RsaKeyPair::from_der(&der).map_err(|err| err.to_string())?;
+        let key_pair = rsa::RsaPrivateKey::from_pkcs1_pem(pem).map_err(|err| err.to_string())?;
```

Comment:
so glad this crate doesn't have to do PEM parsing anymore!!

I think we can remove the `RSA_HEADER` and `RSA_FOOTER` checks now that the `rsa` crate is assuming responsibility for the parsing